### PR TITLE
Remove pillow from dependencies

### PR DESCRIPTION
**Pull request by AleksanderWWW**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [ ] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: tests/unit/neptune/new/attributes/series/test_file_series.py
Comment (szysad) [inline tests/unit/neptune/new/attributes/series/test_file_series.py:1]: why are we marking those tests as `xfail`. Shouldn't we skip if PIL not installed?

File: tests/unit/neptune/new/types/atoms/test_file.py [hunk 1]
```diff
@@ -166,6 +165,8 @@ def test_stream_read_once(self):
             _ = file.content
 
     def test_as_image(self):
+        from PIL import Image
+
```
Comment (szysad) [inline tests/unit/neptune/new/types/atoms/test_file.py:169]: shouldn't PIL be loaded using `pytest.importskip` here too?

Comment (AleksanderWWW) [reply tests/unit/neptune/new/types/atoms/test_file.py:169]: Good point!

**Comment by AleksanderWWW** [comment]
## E2E :hourglass_flowing_sand: 
https://github.com/neptune-ai/neptune-client/actions/runs/8717801814

File: tests/e2e/utils.py
Comment (szysad) [inline tests/e2e/utils.py:1]: whats the reason for getting rid of `PIL importskip` in favor of dynamic import

**Review by szysad — APPROVED** [review]

**Review by szysad — CHANGES_REQUESTED** [review]
One more important thing. Shouldn't `Pillow` be now added to `dev_requirements.txt` so tests that use `Pillow` are not skipped?

File: tests/e2e/utils.py
Comment (AleksanderWWW) [reply tests/e2e/utils.py:1]: `importskip` seems to work only in test files that are not imported later on. Here's a `utils` file that's not directly tested. The result of using `importskip` here was an error:
https://github.com/neptune-ai/neptune-client/actions/runs/8717801814/job/23914764348

**Review by szysad — APPROVED** [review]

**Pull request merged**